### ISSUE TITLE
Nullability mismatch from a switch expression cannot be suppressed directly and must be suppressed on every arm

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#52143]
**Version Used**: 
```
Branch main (25 Mar 2021)
Latest commit e2f6128 by Tomáš Matoušek:
Add RestrictedIVT to dotnet watch to Features (#52087)
```

**Steps to Reproduce**:

Compile the following code
```cs
#nullable enable
C? nullable = null;

C c2 = (args[0].Length switch
{
  0 => nullable,
  _ => null
})!; // CS8600 Converting null literal or possible null value to non-nullable type.
     // single suppression doesn't work; I have to suppress every arm one-by-one

class C { }
```

**Expected Behavior**:
No warnings; The value assigned to `C c2` is under a nullable warnings suppression expression

**Actual Behavior**:
`CS8600 Converting null literal or possible null value to non-nullable type.` is reported for the suppressed switch expression and in order to get rid of it I have to suppress every single problematic arm
